src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -120,7 +109,11 @@ protected override bool ReleaseHandle()
             // to retry, as the descriptor could actually have been closed, been subsequently reassigned, and
             // be in use elsewhere in the process.  Instead, we simply check whether the call was successful.
             int result = Interop.Sys.Close(handle);
-            Debug.Assert(result == 0, $"Close failed with result {result} and error {Interop.Sys.GetLastErrorInfo()}");
+            if (result != 0)
+            {
+                t_lastCloseErrorInfo = Interop.Sys.GetLastErrorInfo();
+                Debug.Fail($"Close failed with result {result} and error {t_lastCloseErrorInfo}");
```

Comment:
I would delete this `Debug.Fail` since it is legitimate to get here.

Reply:
Done. Makes sense to remove it, it was causing the unit tests to fail in debug.